include/common/core/freertos_task.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <array>
+
+#include "FreeRTOS.h"
+#include "task.h"
+
+namespace freertos_task {
+
+template <uint32_t StackDepth, UBaseType_t Priority, typename Entry>
+class FreeRTOSTask {
```

Comment:
let's make this explicitly not copyable and not movable by doing
```
FreeRTOSTask& operator=(FreeRTOSTask&) = delete;
FreeRTOSTask&& operator=(FreeRTOSTask&&) = delete;
FreeRTOSTask(FreeRTOSTask&) = delete;
FreeRTOSTask(FreeRTOSTask&&) = delete;
```
This is because this class at initialization time holds data structures by-value stored in the class and then passes pointers to them to a third-party system; those pointers will become invalid if the class instance changes its location in memory. So we should prevent that from happening.